django/db/models/query.py

```diff
@@ -1981,6 +2025,12 @@ def _fetch_all(self):
         if self._prefetch_related_lookups and not self._prefetch_done:
             self._prefetch_related_objects()
 
+    async def _async_fetch_all(self):
+        if self._result_cache is None:
+            self._result_cache = [result async for result in RawModelIterable(self)]
```

Comment:
Note in passing: 

(Already in play from #14843, but _interesting_, to me at least 😜) 

The `__aiter__` here calls `BaseIterable._async_generator()` ([ln50](https://github.com/django/django/pull/15637/files#diff-d58ef61559dc7af5fdf7b56fee13571a4d2948e784cd608f6afeacf3ac2fb195R50)) which has this nice step: 

```
        while True:
            chunk = await sync_to_async(next_slice)(sync_generator)
            ...
```

... so we (synchronously) fetch chunks of row before mapping them out asynchronously.